workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/SubtreeManager.kt

```diff
@@ -121,17 +124,22 @@ internal class SubtreeManager<PropsT, StateT, OutputT>(
     handler: (ChildOutputT) -> WorkflowAction<PropsT, StateT, OutputT>
   ): ChildRenderingT {
     // Prevent duplicate workflows with the same key.
-    children.forEachStaging {
-      require(!(it.matches(child, key))) {
-        "Expected keys to be unique for ${child.identifier}: key=\"$key\""
+    workflowTracer.trace("CheckingUniqueMatches") {
+      children.forEachStaging {
+        require(!(it.matches(child, key, workflowTracer))) {
+          "Expected keys to be unique for ${child.identifier}: key=\"$key\""
+        }
```

Comment:
Should these checks be debug only? Or toggeable off to delete their runtime cost?

I guess we'll see them show up in traces if they're a problem, so good job adding this